sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/AbstractMobileCenterService.java

```diff
@@ -180,19 +238,72 @@ protected int getTriggerMaxParallelRequests() {
     }
 
     /**
-     * Check if the service is not active: disabled or not started.
+     * Post a command in background.
+     *
+     * @param runnable command.
+     */
+    protected synchronized void post(Runnable runnable) {
+        post(runnable, null, null);
+    }
+
+    /**
+     * Post a command in background.
      *
-     * @return <code>true</code> if the service is inactive, <code>false</code> otherwise.
+     * @param runnable                command.
+     * @param coreDisabledRunnable    optional alternate command if core is disabled.
+     * @param serviceDisabledRunnable optional alternate command if this service is disabled.
+     * @return false if core not configured (no handler ready yet), true otherwise.
      */
-    protected synchronized boolean isInactive() {
-        if (mChannel == null) {
-            MobileCenterLog.error(LOG_TAG, getServiceName() + " service not initialized, discarding calls.");
+    private synchronized boolean post(final Runnable runnable, final Runnable coreDisabledRunnable, final Runnable serviceDisabledRunnable) {
+        if (mHandler == null) {
+            MobileCenterLog.error(LOG_TAG, getServiceName() + " needs to be started before it can be used.");
+            return false;
+        } else {
+            mHandler.post(new Runnable() {
+
+                @Override
+                public void run() {
+                    if (isInstanceEnabled()) {
+                        runnable.run();
+                    } else if (serviceDisabledRunnable != null) {
+                        serviceDisabledRunnable.run();
+                    } else {
+                        MobileCenterLog.info(LOG_TAG, getServiceName() + " service disabled, discarding calls.");
+                    }
+                }
+            }, coreDisabledRunnable);
             return true;
         }
-        if (!isInstanceEnabled()) {
-            MobileCenterLog.info(LOG_TAG, getServiceName() + " service not enabled, discarding calls.");
-            return true;
+    }
+
+    /**
+     * Helper method to handle getter methods in services.
+     *
+     * @param runnable        command to run if service is enabled.
+     * @param future          future to complete value if service if disabled.
+     * @param valueIfDisabled value to complete in future if service is disabled.
+     * @param <T>             getter value type.
+     */
+    protected synchronized <T> void postAsyncGetter(final Runnable runnable, final DefaultSimpleFuture<T> future, final T valueIfDisabled) {
+        Runnable disabledRunnable = new Runnable() {
+
+            @Override
+            public void run() {
+
+                /* Core or service disabled as the same runnable is used for both conditions. */
+                future.complete(valueIfDisabled);
```

Comment:
post rejects the command if called before start as the handler is null, so then we apply default value to commands send before start